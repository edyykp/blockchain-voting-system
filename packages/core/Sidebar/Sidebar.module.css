.container {
  position: fixed;
  z-index: 999;
  width: 100%;
  height: 100%;
  background: var(--gray-900);
  display: grid;
  align-items: center;
  top: -100%;
  left: 0;
  transition: 0.3s ease-in-out;
  opacity: 0;
}

.isOpen {
  opacity: 100% !important;
  top: 0 !important;
}

.icon {
  color: var(--white);
}

.iconContainer {
  position: absolute;
  top: 1.2rem;
  right: 1.5rem;
  background: transparent;
  font-size: 2rem;
  cursor: pointer;
  outline: none;
}

.menuWrapper {
  color: var(--white);
}

.menu {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(6, 80px);
  text-align: center;
  padding: 0;
}

.link {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  text-decoration: none;
  list-style: none;
  transition: 0.2s ease-in-out;
  color: var(--white);
  cursor: pointer;
}

.link:hover {
  color: var(--primary);
  transition: 0.2s ease-in-out;
}

.ctaWrapper {
  display: flex;
  justify-content: center;
}

.cta {
  border-radius: 50px;
  white-space: nowrap;
  padding: 10px 22px;
  color: var(--gray-900);
  outline: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  text-decoration: none;
  text-align: center;
  font-size: 17px;
  width: 150px;
  background: var(--primary);
  color: var(--white);
  border: none;
}
