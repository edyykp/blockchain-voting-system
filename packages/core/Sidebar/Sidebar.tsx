import { FaTimes } from 'react-icons/fa';
import { Link } from 'react-scroll';

import { useMenu } from '@packages/config';
import { Button } from '@packages/components';

import styles from './Sidebar.module.css';

interface SidebarProps {
  isOpen: boolean;
  toggle: () => void;
}

export const Sidebar = ({ isOpen, toggle }: SidebarProps) => {
  const menuItems = useMenu();

  return (
    <aside
      className={`${styles.container} ${isOpen ? `${styles.isOpen}` : ''}`}
      data-testid="sidebar"
    >
      <div className={styles.iconContainer}>
        <FaTimes className={styles.icon} onClick={toggle} />
      </div>
      <div className={styles.menuWrapper}>
        <ul className={styles.menu}>
          {menuItems.map((item, key) => (
            <Link
              to={item.url}
              className={styles.link}
              onClick={toggle}
              key={key}
            >
              {item.title}
            </Link>
          ))}
        </ul>
      </div>
      <div className={styles.ctaWrapper}>
        <Button theme="primary" size="md" />
      </div>
    </aside>
  );
};
