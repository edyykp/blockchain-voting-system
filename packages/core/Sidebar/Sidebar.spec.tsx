import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Sidebar } from './Sidebar';

const mockProps = {
  toggle: () => {},
  isOpen: true,
};

const setup = () => {
  return {
    ...render(<Sidebar {...mockProps} />),
  };
};

describe('Navbar component', () => {
  it('renders', () => {
    const { container, getByTestId } = setup();

    expect(getByTestId('sidebar')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
