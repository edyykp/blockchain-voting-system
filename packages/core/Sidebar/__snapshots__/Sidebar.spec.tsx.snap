// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebarâ†’ renders 1`] = `
<div>
  <aside
    class="container isOpen"
    data-testid="sidebar"
  >
    <div
      class="iconContainer"
    >
      <svg
        class="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </div>
    <div
      class="menuWrapper"
    >
      <ul
        class="menu"
      >
        <a
          class="link"
        >
          About
        </a>
        <a
          class="link"
        >
          Data Protection
        </a>
      </ul>
    </div>
    <div
      class="ctaWrapper"
    >
      <a
        class="button  primary
      false
      mediumButton
      false
      "
      >
        Sign in
      </a>
    </div>
  </aside>
</div>
`;
